dags/datasets/agol-datastore.py

```diff
@@ -0,0 +1,303 @@
+    { "id": "Budget_Type", "type": "text", "info": { "notes": "Approved Budget or Proposed Budget" } },
+    { "id": "Organization_Entity", "type": "text", "info": { "notes": "Organization in TPS" } },
+    { "id": "Command_Name", "type": "text", "info": { "notes": "Name of command" } },
+    { "id": "Pillar_Name", "type": "text", "info": { "notes": "Name of Pillar" } },
+    { "id": "District_Name", "type": "text", "info": { "notes": "District  Name" } },
+    { "id": "Unit_Name", "type": "text", "info": { "notes": "Unit name: Central Paid Duty" } },
+    { "id": "Feature_Category", "type": "text", "info": { "notes": "Category: Revenues" } },
+    { "id": "Cost_Element", "type": "text", "info": { "notes": "Number of cost element" } },
+    { "id": "Cost_Element_Long_Name", "type": "text", "info": { "notes": "Name of cost element: PAID DUTY" } },
+    { "id": "Amount", "type": "float8", "info": { "notes": "Budget amount" } }
+],
+"Budget_by_Command":
+[
+    { "id": "Year", "type": "float8", "info": { "notes": "Fiscal Year" } },
+    { "id": "Type_of_Metric", "type": "text", "info": { "notes": "Type of Metric" } },
+    { "id": "Category", "type": "text", "info": { "notes": "Category" } },
+    { "id": "Command", "type": "text", "info": { "notes": "Command" } },
+    { "id": "Amount", "type": "float8", "info": { "notes": "Budget amount" } },
+],
+"Staffing_by_Command":
+[
+    { "id": "Year", "type": "float8", "info": { "notes": "Fiscal Year" } },
+    { "id": "Type_of_Metric", "type": "text", "info": { "notes": "Type of Metric" } },
+    { "id": "Command", "type": "text", "info": { "notes": "Command" } },
+    { "id": "Category", "type": "text", "info": { "notes": "Category" } },
+    { "id": "Count_", "type": "float8", "info": { "notes": "Budget amount" } },
+],
+}
+common_job_settings = {
+    "description": "Toronto Police Service - budget",
+    "start_date": datetime(2021, 9, 30, 0, 0, 0),
+    "schedule": "@once",
+}
+
+# custom function to create multiple custom dags
+def create_dag(dag_id,
+               dataset,
+               schedule,
+               default_args):
+
+    dag = DAG(dag_id,
+              schedule_interval=schedule,
+              default_args=default_args,
+              tags=['police', 'gary']
+              )
+
+    with dag:
+        # init vars
+        data_filename = "data.json"
+        agol_dataset = dataset["agol_dataset"]
+        package_name=dataset['package_id']
+        resource_name = dataset["name"]
+
+        tmp_dir = CreateLocalDirectoryOperator(
+            task_id = "tmp_dir", 
+            path = TMP_DIR / dag_id
+        )   
+
+        fields_filepath = TMP_DIR / dag_id / "fields.json"
+
+        get_package = GetPackageOperator(
+            task_id="get_package",
+            address=CKAN,
+            apikey=CKAN_APIKEY,
+            package_name_or_id=package_name,
+        )
+
+        def is_resource_new(**kwargs):
```

Comment:
I'm trying again: The function to check the return result is needed for branching

Reply:
Ah, I see. You use that to determine whether you need to backup an existing resource or create a new one. Looks good to me.

A thought: I wonder if we could make an operator to make that process simpler? Would it even be worth is?
 Like - an operator that says "if theres a resource already, Ill back it up, otherwise Ill create a new one"

This comment is nothing you need to address in this PR @garyqi-to dont worry - Im just thinking out loud